src/sidebar/oauth-auth.js

```diff
@@ -264,9 +264,18 @@ function auth($http, $rootScope, $window,
       }
 
       if (Date.now() > token.expiresAt) {
+        var shouldPersist = true;
+
+        // If we are using automatic login via a grant token, do not persist the
+        // initial access token or refreshed tokens.
+        var cfg = serviceConfig(settings);
+        if (cfg && typeof cfg.grantToken !== 'undefined') {
+          shouldPersist = false;
+        }
```

Comment:
`tokenGetter()` is getting pretty long (over 70 lines). It probably needs to be split up very soon, if not overdue. It's doing a lot of different things

1. Decides whether automatic login is in use
2. Exchanges grant token for access token
3. Shows error for expired access token
4. Exchanges auth code for access token
5. Attempts to load tokens from previous session
6. Decides whether tokens should be persisted
7. Attempts to refresh tokens
8. ...

This is probably a sign that `oauth-auth.js` itself has gotten too long and needs to be broken up soon.

Reply:
Hmm. There are three things going on here:

1. Fetch a token from one of three sources (automatic login, auth code, browser storage) if we haven't got one yet.
2. Await the fetched token
3a. Return the fetched token if valid
3b. Otherwise, refresh the token and await the result.

I could split out (1) into separate function but I'd prefer to do that as a separate PR.